_posts/2021-01-27-#20833.md

```diff
@@ -0,0 +1,164 @@
+  relationship, i.e., some transaction spends the outputs of another
+  transaction in the package.  We might call a package a "transaction chain" when
+  it has multiple generations of such dependencies.
+
+- A **test accept** of validation refers to an invocation of mempool validation
+  that validates but does not submit a transaction to the mempool, called from
+  the `testmempoolaccept` RPC.
+
+- **Coins** ([`CCoins`](https://doxygen.bitcoincore.org/struct_c_coin.html))
+  are UTXOs.  In our context, coins don't necessarily need to exist on-chain;
+  they can be outputs from transactions in the mempool or some other valid but
+  unconfirmed transaction.  A **Coins View** is a data structure that contains
+  the available coins. The typical usage is to query a Coins View with a
+  `COutPoint`.
+
+## Background
+
+- The `testmempoolaccept` RPC was introduced in
+  [#11742](https://github.com/bitcoin/bitcoin/pull/11742) and allows users to
+  test raw transactions for validity. While the return object is an array (with
+  the intention to allow multiple transactions), `testmempoolaccept` is currently
+  limited to one transaction at a time, which means users cannot test
+  transactions that depend on outputs that the mempool has not seen yet. The main
+  reason for this limitation is that our validation code only uses the coins
+  available in the current chain state and mempool (and not anything else), to
+  validate transaction inputs.
+
+- Another reason package mempool accept may be of interest is to improve fee
+  bumping.  Currently, Child Pays for Parent does not work for transactions
+  with a feerate below the mempool minimum fee, which can be dangerous for
+  recipients of time-sensitive transactions.
+  [#20833](https://github.com/bitcoin/bitcoin/pull/20833) is a first step towards
+  package validation and hopefully, eventually, [Package
+  Relay](https://bitcoinops.org/en/topics/package-relay/).  We have discussed
+  Package Relay in a previous review club,
+  [#16401](https://bitcoincore.reviews/16401). 
+
+- You can read more about Package Mempool Acceptance
+  [here](https://brink.dev/blog/2021/01/21/fellowship-project-package-accept/).
+
+## Notes
+
+- [#20833](https://github.com/bitcoin/bitcoin/pull/20833) enables dry runs of
+  package validation.  It does this by having `CCoinsViewMemPool` keep track of
+  the Coins created and spent by transactions in the package.
+
+- The
+  [`MemPoolAccept`](https://github.com/bitcoin/bitcoin/blob/7f653c3b/src/validation.cpp#L451)
+  class represents a "validation session" with the current mempool and chain
+  state; it holds `cs_main` and `pool.cs` mutexes throughout validation.  Its job
+  is to figure out whether a transaction (received from peers or clients like
+  RPC) is a valid candidate for submission to the mempool.  To do so, it applies
+  consensus and
+  [policy](https://github.com/bitcoin/bitcoin/tree/c831e105/src/policy) rules.
```

Comment:
I'm not sure if I'd link to this for policy. There are more policy rules than exist in the /src/policy directory.